torch/utils/cpp_extension.py

```diff
@@ -538,6 +543,8 @@ def load_inline(name,
         functions: A list of function names for which to generate function
             bindings. If a dictionary is given, it should map function names to
             docstrings (which are otherwise just the function names).
+        with_cuda: If ``cuda_sources`` is provided, CUDA will always be
+            enabled. Use this flag for pure CPP extensions that require CUDA.
```

Comment:
Hmm, this phrasing is a little weird. Maybe something like "Whether or not to enable CUDA. By default, CUDA is disabled unless `cuda_sources` is provided; you might explicitly set this to True if your extension requires CUDA but doesn't actually have any CUDA source files."

Reply:
@ezyang The thing that is tricky to describe is that with_cuda=False will still give you cuda if there are cuda_souces. Maybe the option would make more sense if it was something like `force_cuda` instead? Let me know if you like the below.

"Whether or not to enable CUDA. If `cuda_sources` is provided, this option is ignored and CUDA is always enabled. You might explicitly set this to True if your extension requires CUDA but doesn't actually have any CUDA source files."

Reply:
Maybe @goldsborough has an API opinion here, but what about defaulting `with_cuda` to `None`, and then always respecting its value when its not None?